docs/PluginAPI.md

```diff
@@ -22,6 +22,14 @@ Calls your callback when the kernel has changed.
 
 * **Function** *Callback* 
 
+## onDidStartKernel(Callback)
+
+Calls your callback when a new kernel starts.
+
+### Params:
+
+* **Function** *Callback* 
```

Comment:
Should probably document the callback signature, stating that it returns a kernel object.

Reply:
The two ideas I have for this are: `* @param {Function} Callback(?kernel)` or `* @param {Function} Callback(kernel|null)`

markdox doesnt seem to like whitespace, this is what the generated markdown looks like with the first option:

![api-docs-callback-param1](https://user-images.githubusercontent.com/10860657/30520814-b0bba332-9b7a-11e7-8501-d9d75625f57f.png)

Reply:
Seems ok to me, I'm certainly no expert there. If someone can fix it later, it'll be available to address. This definitely gives good guidance.

Wow, apparently markdox has never supported documenting callbacks well. https://github.com/cbou/markdox/issues/1

Reply:
I hadnt seen that issue, but was pretty certain that was the case since the [examples](https://github.com/cbou/markdox/blob/master/examples/docs/all.md) seem to assume that all your params will be strings.

I have a couple ideas for this PR next chance I get. If I cant do it simply I may close this, but will still do a separate PR to fix the docs if that is the case.

Reply:
I think it's ok to not worry about perfecting this part, just seemed like a good opportunity to add in a bit more info in the docs.